README.md

```diff
@@ -1,65 +1,76 @@
-	                                   constant:0.0]];
-	    
-	    [self addConstraint:
-	     [NSLayoutConstraint constraintWithItem:self.titleLabel
-	                                  attribute:NSLayoutAttributeTop
-	                                  relatedBy:NSLayoutRelationEqual
-	                                     toItem:self
-	                                  attribute:NSLayoutAttributeTop
-	                                 multiplier:1.0
-	                                   constant:0.0]];
-	}
-
-### Team
-
-* Developers: [Shagun Madhikarmi](mailto:shagun@ustwo.com), [Martin Stolz](mailto:martin@ustwo.com)
\ No newline at end of file
+## Installation
+
+- Add the `BaseView.swift` file to your Xcode project.
+
+## Usage
+
+Use the `BaseView` as a base class for all your custom views. Then override our common setup functions and have them automatically called as part of the initialization.
+
+`BaseView` itself is a subclass of `UIView`. It contains three empty setup functions that are called by both `init(frame:)` and `awakeFromNib()`. These commonly used setup functions are `setup()`, `setupConstraints()`, and `setupAccessibility()` and are called in that order.
+
+## Examples
+
+Here are some example implementations of using the three setup functions in practice. In all of the examples we call `super` on the setup function, but do this as appropriate in your own code.
+
+We use `titleLabel` throughout our examples. We assume that this has been added as a `UILabel` property to the custom view.
+
+### setup
+
+Use the `setup()` function to initialize and subviews, set default values, etc.
+
+```swift
+override func setup() {
+  super.setup()
+
+  backgroundColor = UIColor.redColor()
+
+  titleLabel = UILabel()
+  addSubview(titleLabel)
+}
+```
+
+### setupConstraints
+
+Use the `setupConstraints()` function to layout all the subviews.
+
+```swift
+override func setupConstraints() {
+  super.setupConstraints()
+
+  titleLabel.translatesAutoresizingMaskIntoConstraints = false
+
+  addConstraint(NSLayoutContaint(item: titleLabel, attribute: .Left, relatedBy: .Equal, toItem: self, attribute: .Left, multiplier: 1.0, constant: 0.0))
+  addConstraint(NSLayoutContaint(item: titleLabel, attribute: .Right, relatedBy: .Equal, toItem: self, attribute: .Right, multiplier: 1.0, constant: 0.0))
+  addConstraint(NSLayoutContaint(item: titleLabel, attribute: .Top, relatedBy: .Equal, toItem: self, attribute: .Top, multiplier: 1.0, constant: 0.0))
+}
+```
+
+### setupAccessibility
+
+Use the `setupAccessibility()` function to add any `accessibilityIdentifier` for testing/debugging as well as any `accessibilityLabel` and `accessibilityHint` as appropriate for your users.
+
+```swift
+override func setupAccessibility {
+  super.setupAccessibility()
+
+  titleLabel.accessibilityIdentifier = "TITLE"
+}
+```
+
+## Credits
+
+* Developers: [Shagun Madhikarmi](mailto:shagun@ustwo.com), [Martin Stolz](mailto:martin@ustwo.com)
```

Comment:
We usually use Team for our project templates but maybe Credits or Maintainers is better for OSS 

See here for original discussion 
https://github.com/ustwo/development-at-ustwo/blob/master/git_guidelines.md#readme

and a direct link for for the original template 

https://gist.github.com/alexfish/989853873a762dd4062b

Also feel free to add yourself here if we go with maintainers and if you'd like to